tests/unit/steps/test_plan.py

```diff
@@ -295,8 +298,174 @@ async def test_generate_plan(mock_filter_hypervisors, model, cli_args):
         apps_data_plane=[ceph_osd, nova_compute, ovn_chassis],
     )
 
-    upgrade_plan = await cou_plan.generate_plan(analysis_result, cli_args)
+    upgrade_plan, warning_messages = await cou_plan.generate_plan(analysis_result, cli_args)
     assert str(upgrade_plan) == exp_plan
+    assert len(warning_messages) == 0
+
+
+@pytest.mark.asyncio
+@patch("cou.steps.plan._filter_hypervisors_machines")
+async def test_generate_plan_with_warning_messages(mock_filter_hypervisors, model, cli_args):
```

Comment:
For me these test is testing so many other things, not just warning messages. I'm not sure if these is even usefull.